src/subsidy/subsidy.controller.ts

```diff
@@ -0,0 +1,115 @@
+import { SubsidyParamsDto } from '~/subsidy/dto/subsidy-params.dto';
+import { SubsidyModel } from '~/subsidy/models/subsidy.model';
+import { SubsidyService } from '~/subsidy/subsidy.service';
+
+@Controller('accounts/subsidy')
+export class SubsidyController {
+  constructor(private readonly subsidyService: SubsidyService) {}
+
+  @ApiOperation({
+    summary: 'Get Account Subsidy',
+    description:
+      'The endpoint retrieves the subsidized balance of this Account and the subsidizer Account',
+  })
+  @ApiParam({
+    name: 'subsidizer',
+    description: 'The Account address of the subsidizer',
+    type: 'string',
+    example: '5GwwYnwCYcJ1Rkop35y7SDHAzbxrCkNUDD4YuCUJRPPXbvyV',
+  })
+  @ApiParam({
+    name: 'subsidizer',
+    description: 'The Account address of the beneficiary',
+    type: 'string',
+    example: '5GwwYnwCYcJ1Rkop35y7SDHAzbxrCkNUDD4YuCUJRPPXbvyV',
+  })
+  @ApiOkResponse({
+    description: 'Subsidy details for the Account',
+    type: SubsidyModel,
+  })
+  @ApiNotFoundResponse({
+    description: 'The Subsidy no longer exists',
+  })
+  @Get(':subsidizer/:beneficiary')
+  async getSubsidy(@Param() { beneficiary, subsidizer }: SubsidyParamsDto): Promise<SubsidyModel> {
+    const allowance = await this.subsidyService.getAllowance(beneficiary, subsidizer);
+
+    return new SubsidyModel({
+      beneficiary: new AccountModel({ address: beneficiary }),
+      subsidizer: new AccountModel({ address: subsidizer }),
+      allowance,
+    });
+  }
+
+  @ApiOperation({
+    summary: 'Subsidize an account',
+    description:
+      'This endpoint sends an Authorization Request to an Account to subsidize its transaction fees',
+  })
+  @ApiTransactionResponse({
+    description: 'Newly created Authorization Request along with transaction details',
+    type: CreatedAuthorizationRequestModel,
+  })
+  @ApiTransactionFailedResponse({
+    [HttpStatus.BAD_REQUEST]: [
+      'The Beneficiary Account already has a pending invitation to add this account as a subsidizer with the same allowance',
+    ],
+  })
+  @Post('create')
+  async subsidizeAccount(@Body() params: CreateSubsidyDto): Promise<TransactionResponseModel> {
+    const serviceResult = await this.subsidyService.subsidizeAccount(params);
+
+    return handleServiceResult(serviceResult, authorizationRequestResolver);
+  }
+
+  @ApiOperation({
+    summary: 'Modify allowance for a Subsidy relationship an account',
+    description:
+      'This endpoint allows to set/increase/decrease allowance of a Subsidy relationship. Note that only the subsidizer is allowed to modify the allowance',
+  })
+  @ApiTransactionResponse({
+    description: 'Details about the transaction',
+    type: TransactionQueueModel,
+  })
+  @ApiTransactionFailedResponse({
+    [HttpStatus.BAD_REQUEST]: [
+      'The Beneficiary Account already has a pending invitation to add this account as a subsidizer with the same allowance',
+    ],
+  })
+  @Post('allowance/modify')
```

Comment:
I think this should be 3 endpoints instead of nesting method dispatching in the body. Its more boiler plate and documentation, but is more consistent being broken up with one route, one method.

The service method can stay the same and the operation can be hard coded in the controller function in the call